src/read.c

```diff
@@ -4401,31 +4410,15 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         }
     }
     assert(alphaItemCount == colorItemCount);
-    // Find an unused ID.
-    avifResult result;
-    if (meta->items.count >= UINT32_MAX - 1) {
-        // In the improbable case where all IDs are used.
-        result = AVIF_RESULT_DECODE_ALPHA_FAILED;
-    } else {
-        uint32_t newItemID = 0;
-        avifBool isUsed;
-        do {
-            ++newItemID;
-            isUsed = AVIF_FALSE;
-            for (uint32_t i = 0; i < meta->items.count; ++i) {
-                if (meta->items.item[i]->id == newItemID) {
-                    isUsed = AVIF_TRUE;
-                    break;
-                }
-            }
-        } while (isUsed && newItemID != 0);
-        result = avifMetaFindOrCreateItem(meta, newItemID, alphaItem); // Create new empty item.
-    }
-    if (result != AVIF_RESULT_OK) {
+    // Create a temporary item;
+    avifCreateItem(alphaItem);
```

Comment:
IMPORTANT: Check whether `*alphaItem` is a null pointer.

It seems that line 4416 is meant to check that, but the check is incorrect (it should check `*alphaItem`, not  `alphaItem`) and too late (`*alphaItem` is already dereferenced at line 4415).

Reply:
Outdated code